### PR TITLE
refactor(all): Redundant throws [done by IDE]

### PR DESCRIPTION
**Pull request by jamdavi**
I used the IntelliJ tool to remove all of these redundant throws. There are a number of public methods that got changed and if this is not desirable I can revert this and do it one by one.

### CODE REVIEW

File: device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/DeviceClient.java [hunk 1]
```diff
@@ -575,9 +575,8 @@ public void uploadToBlobAsync(String destinationBlobName, InputStream inputStrea
      * @param request The request details for getting the SAS URI, including the destination blob name.
      * @return The file upload details to be used with the Azure Storage SDK in order to upload a file from this device.
      * @throws IOException If this HTTPS request fails to send.
-     * @throws URISyntaxException If the returned sas uri cannot be constructed correctly
      */
-    public FileUploadSasUriResponse getFileUploadSasUri(FileUploadSasUriRequest request) throws IOException, URISyntaxException
```
Comment (timtay-microsoft) [inline device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/DeviceClient.java:580]: This would be a breaking change since this is a public method. The best we can do here is suppress the warning and note in the javadocs that URISyntaxException is never thrown

File: device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/ModuleClient.java [hunk 2]
```diff
@@ -170,7 +167,7 @@ public ModuleClient(String connectionString, IotHubClientProtocol protocol, Stri
      * See <a href="https://github.com/Azure/azure-iot-sdk-java/blob/master/device/iot-device-samples/send-event-x509/pom.xml">this pom.xml</a> for an example of how to do this.
      */
     @Deprecated
-    public ModuleClient(String connectionString, IotHubClientProtocol protocol, SSLContext sslContext) throws ModuleClientException, URISyntaxException
```
Comment (timtay-microsoft) [inline device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/ModuleClient.java:173]: Same comment for these ModuleClientExceptions. These would be breaking changes to remove

File: provisioning/security/security-provider/src/main/java/com/microsoft/azure/sdk/iot/provisioning/security/SecurityProviderX509.java [hunk 3]
```diff
@@ -27,7 +27,7 @@
     abstract public Collection<X509Certificate> getIntermediateCertificatesChain();
 
     @Override
-    public String getRegistrationId() throws SecurityProviderException
```
Comment (timtay-microsoft) [inline provisioning/security/security-provider/src/main/java/com/microsoft/azure/sdk/iot/provisioning/security/SecurityProviderX509.java:30]: This would also be a breaking change since users do call this method directly under some circumstances

File: provisioning/security/tpm-provider/src/main/java/com/microsoft/azure/sdk/iot/provisioning/security/hsm/SecurityProviderTPMHsm.java [hunk 4]
```diff
@@ -393,10 +392,9 @@ else if (rc != TPM_RC.HANDLE)
     /**
      * Getter for extracting StorageRootKey from TPM
      * @return The StorageRootKey from TPM
-     * @throws SecurityProviderException If StorageRootKey could not be extracted.
      */
     @Override
-    public byte[] getStorageRootKey() throws SecurityProviderException
```
Comment (timtay-microsoft) [inline provisioning/security/tpm-provider/src/main/java/com/microsoft/azure/sdk/iot/provisioning/security/hsm/SecurityProviderTPMHsm.java:399]: Same comment for these two methods. They are typically only used internally, but this class is directly used by customers, so it would risk breaking them

File: provisioning/security/tpm-provider-emulator/src/main/java/com/microsoft/azure/sdk/iot/provisioning/security/hsm/SecurityProviderTPMEmulator.java [hunk 5]
```diff
@@ -511,10 +511,9 @@ else if (rc != TPM_RC.HANDLE)
     /**
      * Getter for extracting EndorsementKey from TPM
      * @return The Endorsement Key from TPM
-     * @throws SecurityProviderException If endorsement key could not be extracted.
      */
     @Override
-    public byte[] getEndorsementKey() throws SecurityProviderException
```
Comment (timtay-microsoft) [inline provisioning/security/tpm-provider-emulator/src/main/java/com/microsoft/azure/sdk/iot/provisioning/security/hsm/SecurityProviderTPMEmulator.java:517]: This class is arguably only for testing purposes, and customers aren't expected to use it outside of their own tests, but we may want to not break it just to be safe

File: service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/IotHubConnectionStringBuilder.java [hunk 6]
```diff
@@ -26,12 +26,10 @@
      *
      * @param connectionString The connection string
      * @return The IotHubConnectionString object
-     * @throws IOException This exception is never thrown. Users are advised to use {@link #createIotHubConnectionString(String, AuthenticationMethod)} instead
-     * since it's method signature does not declare any thrown exceptions.
      * @deprecated Use {@link #createIotHubConnectionString(String)} since it does not declare any thrown exceptions
      */
     @Deprecated
-    public static IotHubConnectionString createConnectionString(String connectionString) throws IOException
```
Comment (timtay-microsoft) [inline service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/IotHubConnectionStringBuilder.java:34]: This class is about 1 layer removed from being public, so I'd revert this change. We reference it in a sample or two when demonstrating how to build SAS tokens

File: service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/Message.java [hunk 7]
```diff
@@ -210,9 +210,8 @@ public Message(byte[] byteArray)
      * Important: If a string is passed, the HttpBatch.SerializeAsString is set to true,
      * and the internal byte representation is serialized as UTF-8,
      * with HttpBatch.Encoding set to UTF-8.
-     * @throws UnsupportedEncodingException This exception is thrown if unsupported encoding used
      */
-    public Message(String string) throws UnsupportedEncodingException
```
Comment (timtay-microsoft) [inline service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/Message.java:215]: Users do use this API when sending c2d messages, so this would be a breaking change

File: service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/RegistryManager.java [hunk 8]
```diff
@@ -102,12 +102,11 @@ public static RegistryManager createFromConnectionString(String connectionString
      * @deprecated because this method declares a thrown IOException even though it never throws an IOException. Users
      * are recommended to use {@link #RegistryManager(String, RegistryManagerOptions)} instead since it does not declare this exception even
      * though it constructs the same RegistryManager.
-     * @throws IOException This exception is never thrown.
      */
     @Deprecated
     public static RegistryManager createFromConnectionString(
             String connectionString,
-            RegistryManagerOptions options) throws IOException
```
Comment (timtay-microsoft) [inline service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/RegistryManager.java:110]: Very publicly used method, so this would be a breaking change for a lot of users

Comment (timtay-microsoft) [reply service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/RegistryManager.java:110]: Same for the public methods in this class

**Review by timtay-microsoft — CHANGES_REQUESTED** [review]
Lots of breaking changes in here. The client level objects in particular should not remove these exceptions, even though we don't throw them.

**Comment by jamdavi** [comment]
Not going to handle this scenario as most of these are public methods and changing the signature would break a lot of things. Will add this inspection to the ignore list
